src/Roslyn.Suppressions/Roslyn.Suppressions/RoslynSettings.cs

```diff
@@ -75,26 +67,26 @@ public SuppressedIssue(string filePath, string hash, RoslynLanguage roslynLangua
         /// The path is in Windows format i.e. the directory separators are backslashes
         /// </remarks>
         [JsonProperty("file")]
-        public string FilePath { get; }
+        public string FilePath { get; set; }
 
         [JsonProperty("hash")]
-        public string Hash { get; }
+        public string Hash { get; set; }
 
         [JsonProperty("lang")]
-        public RoslynLanguage RoslynLanguage { get; }
+        public RoslynLanguage RoslynLanguage { get; set; }
 
         /// <summary>
         /// The rule ID reported by the Roslyn analyzer e.g. S123
         /// </summary>
         /// <remarks>This Sonar rule key without repository key.</remarks>
         [JsonProperty("rule")]
-        public string RoslynRuleId { get; }
+        public string RoslynRuleId { get; set; }
 
         /// <summary>
         /// The 0-based line for the issue. Will be null for file-level issues
         /// </summary>
         /// <remarks>Roslyn issues are 0-based - Sonar issues are 1-based.</remarks>
         [JsonProperty("line")]
-        public int? RoslynIssueLine { get; }
+        public int? RoslynIssueLine { get; set; }
```

Comment:
NewtonSoft does have settings to configure how nulls are handled,  but they didn't fix the issue.